ide/code-editor.js

```diff
@@ -64,17 +65,16 @@ export default class CodeEditor extends Morph {
   
   highlight() {
     if (!this.theme) return;
-    const txt = this.submorphs[0],
-          tokens = this.mode.highlight(txt.textString),
-          defaultStyle = this.submorphs[0].styleProps,
-          styleRanges = tokens.map(({token, from, to}) =>
-            StyleRange.fromPositions({...defaultStyle, ...this.theme.styleCached(token)}, from, to));
-    styleRanges.push(StyleRange.create(defaultStyle, 0, -1, 0, 0));
-    txt.replaceStyleRanges(styleRanges);
-
-    if (this._checker) {
+    let textMorph = this.submorphs[0],
+        tokens = this.mode.highlight(textMorph.textString),
+        defaultStyle = this.submorphs[0].styleProps;
+    let textAttributes = tokens.map(({token, from, to}) =>
+          TextAttribute.fromPositions({...defaultStyle, ...this.theme.styleCached(token)}, from, to));
```

Comment:
This creates a lot of new objects and might cause GC pressure, so if syntax highlighting is still slow, this line could probably be optimized (especially as long as `defaultStyle` is empty).